src/Uno.UI/UI/Xaml/Controls/ProgressRing/ProgressRing.wasm.cs

```diff
@@ -1,22 +1,63 @@
 ﻿#if __WASM__
 using System;
-using System.Collections.Generic;
-using System.Text;
-using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Controls.Primitives;
 
 namespace Windows.UI.Xaml.Controls
 {
-    public partial class ProgressRing : Control
+	public partial class ProgressRing : Control
 	{
 		private static void OnForegroundChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
 		{
-			
+
 		}
 
 		private static void OnIsActiveChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
 		{
-		
+			var progressRing = dependencyObject as ProgressRing;
+			var isActive = args.NewValue as bool?;
+
+			if (progressRing != null && progressRing.IsLoaded && isActive.HasValue)
+			{
+				VisualStateManager.GoToState(progressRing, isActive.Value ? "Active" : "Inactive", false);
+			}
+		}
+
+		protected override void OnLoaded()
+		{
+			base.OnLoaded();
+			// The initial call to OnIsActiveChanged fires before ProgressRing is Loaded, so we also need to set a proper VisualState here
+			VisualStateManager.GoToState(this, IsActive ? "Active" : "Inactive", false);
 		}
+
+		public ProgressRingTemplateSettings TemplateSettings
+		{
+			get
+			{
+				var result = new ProgressRingTemplateSettings()
+				{
+					EllipseDiameter = 3,
```

Comment:
Where are these values from?

Reply:
That's ok for now, please create an issue to track it and include the issue # in the comment.

Reply:
Done, #1872